.github/workflows/kernel-checks.yml

```diff
@@ -9,10 +9,7 @@ jobs:
     steps:
       # Install python 3
       - name: Tool Setup
-        uses: actions/setup-python@v2
-        with:
-          python-version: 3.7.10
-          architecture:   x64
+        uses: actions/setup-python@v3
```

Comment:
Should we not set python version/architecture? Obv the default must work... so this is a nitpick more than anything. I guess I just worry that we'll update to a later version and this suddenly breaks if the newer version has a different default.

Reply:
You're right being forced to a new version could lead to a breaking script

Much in the way that being tied to an old version that is deprecated and will be dropped could lead to a breaking script
```
The following actions uses node12 which is deprecated and will be forced to run on node16: actions/setup-python@v2, lots0logs/gh-action-get-changed-files@2.1.4. For more info: https://github.blog/changelog/2023-06-13-github-actions-all-actions-will-run-on-node16-instead-of-node12-by-default/
```

Reply:
Right but can you not specify the python-version field as was done before for setup-python@v2?

Reply:
Not that version.